v23.1/performance-recipes.md

```diff
@@ -77,15 +84,113 @@ This section provides solutions for common performance issues in your applicatio
+Identify the transactions that are in conflict, and unblock them if possible. In general, take steps to [reduce transaction contention](#reduce-transaction-contention).
+
+In addition, implement [client-side retry handling](transaction-retry-error-reference.html#client-side-retry-handling) so that your application can respond to [transaction retry errors](transaction-retry-error-reference.html) that are emitted when CockroachDB cannot [automatically retry](transactions.html#automatic-retries) a transaction.
+
+##### Identify conflicting transactions
+
+- In the **Active Executions** table on the **Transactions** page ([{{ site.data.products.db }} Console](../cockroachcloud/transactions-page.html) or [DB Console](ui-transactions-page.html#active-executions-table)), look for a **waiting** transaction (`Waiting` status).
+  {{site.data.alerts.callout_success}}
+  If you see many waiting transactions, a single long-running transaction may be blocking transactions that are, in turn, blocking others. In this case, sort the table by **Time Spent Waiting** to find the transaction that has been waiting for the longest amount of time. Unblocking this transaction may unblock the other transactions.
+  {{site.data.alerts.end}}
+  Click the transaction's execution ID and view the following transaction execution details:
+  <img src="{{ 'images/v23.1/waiting-transaction.png' | relative_url }}" alt="Movr rides transactions" style="border:1px solid #eee;max-width:100%" />
+  - **Last Retry Reason** shows the last [transaction retry error](#transaction-retry-error) received for the transaction, if applicable.
+  - The details of the **blocking** transaction, directly below the **Contention Insights** section. Click the blocking transaction to view its details.
+
+##### Cancel a blocking transaction
+
+1. [Identify the **blocking** transaction](#identify-conflicting-transactions) and view its transaction execution details.
+1. Click its **Session ID** to open the **Session Details** page.
+  <img src="{{ 'images/v23.1/ui-sessions-details-page.png' | relative_url }}" alt="Sessions Details Page" style="border:1px solid #eee;max-width:100%" />
+1. Click **Cancel Statement** to cancel the **Most Recent Statement** and thus the transaction, or click **Cancel Session** to cancel the session issuing the transaction.
+
+##### Identify transactions and objects that experienced lock contention
+
+To identify transactions that experienced [lock contention](performance-best-practices-overview.html#transaction-contention) in the past:
+
+- In the **Transaction Executions** view on the **Insights** page ([{{ site.data.products.db }} Console](../cockroachcloud/insights-page.html#transaction-executions-view) and [DB Console](ui-insights-page.html#transaction-executions-view)), look for a transaction with the **High Contention** insight. Click the transaction's execution ID and view the transaction execution details, including the details of the blocking transaction.
```

Comment:
Is this information duplicative of what's above?

Reply:
A bit ... but applied in slightly different contexts, so I think I'll let this go. Good observation though.